client/client.go

```diff
@@ -44,27 +42,27 @@ func doScrape(request *http.Request, client *http.Client, logger log.Logger) {
 	scrapeResp, err := client.Do(request)
 	if err != nil {
 		msg := fmt.Sprintf("Failed to scrape %s: %s", request.URL.String(), err)
-		level.Warn(logger).Log(msg)
+		level.Warn(logger).Log("msg", msg, "scrap_id", request.Header.Get("id"))
 		resp := &http.Response{
 			StatusCode: 500,
 			Header:     http.Header{},
 			Body:       ioutil.NopCloser(strings.NewReader(msg)),
 		}
 		err = doPush(resp, request, client)
 		if err != nil {
-			level.Warn(logger).Log("Failed to push failed scrape response:", err)
+			level.Warn(logger).Log("msg", "Failed to push failed scrape response:", err, "scrap_id", request.Header.Get("id"))
 			return
 		}
-		level.Info(logger).Log("Pushed failed scrape response")
+		level.Info(logger).Log("msg", "Pushed failed scrape response", "scrap_id", request.Header.Get("id"))
 		return
 	}
-	level.Info(logger).Log("Retrieved scrape response")
+	level.Info(logger).Log("msg", "Retrieved scrape response", "scrap_id", request.Header.Get("id"))
 	err = doPush(scrapeResp, request, client)
 	if err != nil {
-		level.Warn(logger).Log("Failed to push scrape response:", err)
+		level.Warn(logger).Log("msg", "Failed to push scrape response:", err, "scrap_id", request.Header.Get("id"))
 		return
 	}
-	level.Info(logger).Log("Pushed scrape result")
+	level.Info(logger).Log("msg", "Pushed scrape result", "scrap_id", request.Header.Get("id"))
```

Comment:
Given all these logs have the same scrape_id field, it'd be simpler to use With on the logger to add it